docs/DafnyRef/GrammarDetails.md

```diff
@@ -1414,7 +1417,7 @@ LiteralExpression =
 
 Nat = ( digits | hexdigits )
 
-Dec = decimaldigits
+Dec = ( realnumber | digits "." | "." ( digits | realnumber ) )
```

Comment:
This is a good way to write it for the reference manual. That is, there is no need to complicate things here by saying that no whitespace is allowed between the digits and the trailing/leading dot -- a reader of the reference manual wouldn't expect such space to be allowed anyhow.